src/mds/MDSRank.cc

```diff
@@ -1471,10 +1471,12 @@ void MDSRank::send_message_mds(const ref_t<Message>& m, const entity_addrvec_t &
   messenger->send_to_mds(ref_t<Message>(m).detach(), addr);
 }
 
-void MDSRank::forward_message_mds(const cref_t<MClientRequest>& m, mds_rank_t mds)
+void MDSRank::forward_message_mds(MDRequestRef& mdr, mds_rank_t mds)
```

Comment:
changes look good but do we really need this line? The only use of this seems to be `dout(1) << "session was removed, failed to forward message " << mdr << dendl;` right?

Reply:
Just want to print the request detail when dropping it. BTW, do you have better ideas ?

Reply:
maybe use `m->get_op()`, like 
```
dout(1) << "session was removed, failed to perform " << ceph_mds_op_name(m->get_op()) << dendl;
```

Reply:
we can also use `m->get_caller_uid()` to help debug further

Reply:
While the `mdr` will give more detail info instead of only the op type and uid. Such from which client and the time stamp, etc.